### PR TITLE
feat(helm): add gcEnable configuration to chart for PR #167

### PR DESCRIPTION
**Pull request by njuptlzf**
## Pull Request

**Why is this PR required? What issue does it fix?**:
for PR #167

**What this PR does?**:
add gcEnable configuration to helm chart.

**Does this PR require any upgrade changes?**:

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Any additional information for your reviewer?** : 
_Mention if this PR is part of any design or a continuation of previous PRs_
```yaml
nfsProvisioner:
  # Provide a switch to turn off the function of clearing stale pvc to avoid
  # garbage collecting an NFS backend PVC if the NFS PVC is deleted.
  enableGarbageCollection: "true"
```

### CODE REVIEW

**Comment by njuptlzf** [comment]
@niladrih ptal. :)

**Comment by njuptlzf** [comment]
I found that CI reported an error when executing `ct lint --config ct.yml`
```sh
Run ct lint --config ct.yml
... ...
Error: failed linting charts: failed processing charts
Error: unknown flag: --timeout
------------------------------------------------------------------------------------------------------------------------
 ✖︎ nfs-provisioner => (version: "0.10.3", path: "deploy/helm/charts") > failed waiting for process: exit status 1
------------------------------------------------------------------------------------------------------------------------
failed linting charts: failed processing charts
Error: Process completed with exit code 1.
```
It's strange that `--timeout` is not supported, this is a capability that has been supported since 2021 and is still supported in `helm v3.13.2`, even if CI is using `helm v3.12.1`.

It doesn't happen locally.
```sh
root@k3s-1:/home/go_project/src/github.com/njuptlzf/dynamic-nfs-provisioner# helm version
version.BuildInfo{Version:"v3.13.2", GitCommit:"2a2fb3b98829f1e0be6fb18af2f6599e0f4e8243", GitTreeState:"clean", GoVersion:"go1.20.10"}

root@k3s-1:/home/go_project/src/github.com/njuptlzf/dynamic-nfs-provisioner# ct version
Version:         v3.10.0
Git commit:      0cb17e5aa89e2d6cf49cb4e7f09b602af58adfbb
Date:            2023-10-31T14:27:21Z
License:         Apache 2.0

root@k3s-1:/home/go_project/src/github.com/njuptlzf/dynamic-nfs-provisioner# export CT_CONFIG_DIR=/home/helm/ct-v3.10.0/etc

root@k3s-1:/home/go_project/src/github.com/njuptlzf/dynamic-nfs-provisioner# ct lint --config ct.yml
Linting charts...

------------------------------------------------------------------------------------------------------------------------
 Charts to be processed:
------------------------------------------------------------------------------------------------------------------------
 nfs-provisioner => (version: "0.10.3", path: "deploy/helm/charts")
------------------------------------------------------------------------------------------------------------------------

Linting chart "nfs-provisioner => (version: \"0.10.3\", path: \"deploy/helm/charts\")"
Checking chart "nfs-provisioner => (version: \"0.10.3\", path: \"deploy/helm/charts\")" for a version bump...
Old chart version: 0.10.2
New chart version: 0.10.3
Chart version ok.
Validating /home/go_project/src/github.com/njuptlzf/dynamic-nfs-provisioner/deploy/helm/charts/Chart.yaml...
Validation success! 👍
Validating maintainers...
==> Linting deploy/helm/charts

1 chart(s) linted, 0 chart(s) failed

------------------------------------------------------------------------------------------------------------------------
 ✔︎ nfs-provisioner => (version: "0.10.3", path: "deploy/helm/charts")
------------------------------------------------------------------------------------------------------------------------
All charts linted successfully
root@k3s-1:/home/go_project/src/github.com/njuptlzf/dynamic-nfs-provisioner# 
```

@niladrih, I'm still troubleshooting the differences between CI and local, do you have any ideas? :(

File: deploy/helm/charts/templates/deployment.yaml [hunk 1]
```diff
@@ -103,6 +103,10 @@ spec:
             - name: OPENEBS_IO_NFS_SERVER_NODE_AFFINITY
               value: "{{ .Values.nfsProvisioner.nfsServerNodeAffinity }}"
             {{- end }}
+            # Provide a switch to turn off the function of clearing stale pvc to avoid
+            # garbage collecting an NFS backend PVC if the NFS PVC is deleted.
+            - name: OPENEBS_IO_NFS_SERVER_GARBAGE_COLLECTION_ENABLED
+              value: "{{ .Values.nfsProvisioner.enableGarbageCollection }}"
```
Comment (niladrih) [inline deploy/helm/charts/templates/deployment.yaml:109]: nitpick:
```suggestion
            {{ if .Values.nfsProvisioner.enableGarbageCollection }}
            - name: OPENEBS_IO_NFS_SERVER_GARBAGE_COLLECTION_ENABLED
              value: {{ quote .Values.nfsProvisioner.enableGarbageCollection }}
            {{- end }}
```
And on the values.yaml, the option wouldn't need quotes around it anymore
```yaml
enableGarbageCollection: true
```

File: deploy/helm/charts/README.md [hunk 2]
```diff
@@ -131,6 +131,7 @@ helm install openebs-nfs openebs-nfs/nfs-provisioner --namespace openebs --creat
 | `nfsProvisioner.nfsServerNodeAffinity`       | NFS Server node affinity rules                | `""`                        |
 | `nfsProvisioner.nfsBackendPvcTimeout`       | Timeout for backend PVC binding in seconds                | `"60"`                      |
 | `nfsProvisioner.nfsHookConfigMap`       | Existing Configmap name to load hook configuration                | `""`                        |
+| `nfsProvisioner.enableGarbageCollection`       | Enable garbage collection            | `true`                      |
```
Comment (niladrih) [inline deploy/helm/charts/README.md:134]:
```suggestion
| `nfsProvisioner.enableGarbageCollection`       | Enable garbage collection for the backend PVC | `true`                      |
```

**Review by niladrih — COMMENTED** [review]
Left few comments @njuptlzf

**Comment by niladrih** [comment]
> @niladrih, I'm still troubleshooting the differences between CI and local, do you have any ideas? :(

This is because of this bug https://github.com/helm/chart-testing-action/issues/135#issuecomment-1799004311. It was fixed in 2.6.1 of the action, I'll raise a PR to fix this. I'll pull in your commit and add it my suggested change. I think this will be a release PR as well :)

**Comment by niladrih** [comment]
Closing this, as these changes are included in #178

**Comment by njuptlzf** [comment]
> Left few comments @njuptlzf

Thanks, this will benefit my habits a lot

**Pull request closed without merging**